substrate/frame/revive/src/pure_precompiles/ecrecover.rs

```diff
@@ -0,0 +1,60 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use super::Precompile;
+use crate::{exec::ExecResult, Config, ExecReturnValue, GasMeter, RuntimeCosts};
+use hex_literal::hex;
+use pallet_revive_uapi::ReturnFlags;
+use sp_core::H160;
+pub const ECRECOVER: H160 = H160(hex!("0000000000000000000000000000000000000001"));
+
+/// The ecrecover precompile.
+pub struct ECRecover;
+
+impl<T: Config> Precompile<T> for ECRecover {
+	fn execute(gas_meter: &mut GasMeter<T>, i: &[u8]) -> ExecResult {
+		gas_meter.charge(RuntimeCosts::EcdsaRecovery)?;
```

Comment:
This is correct. But we also need to not charge the contract call gas in case of a pre-compile. Otherwise it will be super expensive since the contract call gas does include accessing the contract code storage item.

Reply:
Yes. Just need to add the same thing to delegate call then we good.